agent/structs/config_entry_gateways.go

```diff
@@ -769,6 +770,9 @@ func (e *APIGatewayConfigEntry) Validate() error {
 		return err
 	}
 
+	if len(e.Listeners) == 0 {
+		return errors.New("api gateway must have at least one listener")
```

Comment:
minor (and more stylistic than anything and not blocking), would we want to use a sentinel error here so callers could act on the error type in the future?

Reply:
So I don't feel like that matches the way the other validations are handled, but I'm also realizing I didn't follow that way either. Ex: https://github.com/hashicorp/consul/blob/main/agent/structs/config_entry_gateways.go#L803

Reply:
Especially because I think it sends a 500 error back right now